### PR TITLE
Fix a string telescoping test in t.anal/x86/emu

### PR DESCRIPTION
**Pull request by MaskRay**

### CODE REVIEW

File: libr/util/p_format.c [hunk 1]
```diff
@@ -349,12 +349,11 @@ static void r_print_format_decchar(const RPrint* p, int endian, int mode,
 
 static int r_print_format_string(const RPrint* p, ut64 seeki, ut64 addr64, ut64 addr, int is64, int mode) {
 	ut8 buffer[255];
+	bool can_read;
```
Comment (radare) [inline libr/util/p_format.c]: Initialize to false

File: libr/util/p_format.c [hunk 2]
```diff
@@ -364,7 +363,9 @@ static int r_print_format_string(const RPrint* p, ut64 seeki, ut64 addr64, ut64
 	} else if (MUSTSEE) {
 		if (!SEEVALUE) p->cb_printf ("0x%08"PFMT64x" = ", seeki);
 		if (!SEEVALUE) p->cb_printf ("0x%08"PFMT64x" -> 0x%08"PFMT64x" ", seeki, addr);
```
Comment (radare) [inline libr/util/p_format.c]: Add braces here

Comment (MaskRay) [reply libr/util/p_format.c]: Added

File: libr/util/p_format.c (hunk 1 above)
Comment (MaskRay) [reply libr/util/p_format.c]: There is no path where `can_read` is not assigned.

Comment (MaskRay) [reply libr/util/p_format.c]: This file is already a hell, as @alvarofe said.

**Pull request closed without merging**